src/dune_engine/fs_memo.ml

```diff
@@ -256,7 +293,7 @@ let invalidate_path_and_its_parent path =
 let handle_fs_event ({ kind; path } : Dune_file_watcher.Fs_memo_event.t) :
     Memo.Invalidation.t =
   match kind with
-  | File_changed -> invalidate_path path
+  | File_changed -> invalidate_path_both path
```

Comment:
This seems wrong: a `File_changed` event is only relevant for files, and files are watched `Directly`. `Via_parent` is used only for directories.

Reply:
(Edited the above comment, confused `Via_parent` and `Directly` at first.)

Reply:
I'll switch to these names.